source/grid/grid_in.cc

```diff
@@ -357,22 +337,20 @@ void GridIn<dim, spacedim>::read_vtk(std::istream &in)
 
           if (dim == 3)
             {
-              for (unsigned int i = 0; i < no_quads; i++) //assigning IDs to bounds.
+              for (unsigned int i = 0; i < n_quads; i++) //assigning IDs to bounds.
                 {
                   double id;
                   in >> id;
-                  Assert (quad_indices[i] < subcelldata.boundary_quads.size(), ExcInternalError());
```

Comment:
I don't think that this and the other Assert will ever trigger, but I think that the logic that they guard is complicated enough that we should keep them.